File: android/app/src/main/java/com/aitutor/app/domain/model/Speaking.kt
package com.aitutor.app.domain.model

import kotlinx.serialization.Serializable

/** Speaking practice modes — mirror the OpenAI Realtime instructions in the proxy. */
enum class SpeakingMode(val displayName: String, val description: String, val wire: String) {
    ACADEMIC(
        "Academic",
        "Formal practice for IELTS Speaking parts 2/3 and TOEFL Independent Speaking",
        "academic"
    ),
    CASUAL(
        "Casual",
        "Friendly chat for everyday fluency practice",
        "native"
    )
}

@Serializable
data class SpeakingTurn(
    val role: Role,
    val content: String
) {
    enum class Role { USER, ASSISTANT }
}

File: ios/AITutor/Domain/Model/Speaking.swift
import Foundation

/// Speaking practice modes — mirror the OpenAI Realtime instructions in the proxy.
enum SpeakingMode: String, CaseIterable, Identifiable {
    case academic
    case casual = "native"

    var id: String { rawValue }

    var displayName: String {
        switch self {
        case .academic: return "Academic"
        case .casual: return "Casual"
        }
    }

    var modeDescription: String {
        switch self {
        case .academic:
            return "Formal practice for IELTS Speaking parts 2/3 and TOEFL Independent Speaking"
        case .casual:
            return "Friendly chat for everyday fluency practice"
        }
    }

    var wire: String { rawValue }
}

struct SpeakingTurn: Equatable, Identifiable {
    enum Role: String { case user, assistant }

    let id: String
    let role: Role
    let content: String

    init(id: String = UUID().uuidString, role: Role, content: String) {
        self.id = id
        self.role = role
        self.content = content
    }
}
